test/functional/feature_rare_output_scripts.py

```diff
@@ -0,0 +1,159 @@
+        for tx in (tx_even, tx_odd):
+            decode_res = self.node.decoderawtransaction(tx['hex'])
+            assert_equal(decode_res['vout'][tx['sent_vout']]['scriptPubKey']['type'], "pubkey")
+
+        self.log.info("Check that spending hybrid pubkey P2PK outputs is non-standard")
+        spend_even_tx = self.create_p2pk_spend((tx_even['txid'], tx_even['sent_vout']), pubkey_even_script, privkey_even)
+        spend_odd_tx = self.create_p2pk_spend((tx_odd['txid'], tx_odd['sent_vout']), pubkey_odd_script, privkey_odd)
+        for tx in (spend_even_tx, spend_odd_tx):
+            assert_raises_rpc_error(-26, "non-mandatory-script-verify-flag (Public key is neither compressed or uncompressed)",
+                                    self.node.sendrawtransaction, tx.serialize().hex())
+
+        self.log.info("Check that spending hybrid pubkey P2PK outputs is consensus-valid")
+        self.generate(self.node, 1)
+        tmpl = self.node.getblocktemplate(NORMAL_GBT_REQUEST_PARAMS)
+        block = create_block(tmpl=tmpl, txlist=[spend_even_tx, spend_odd_tx])
+        block.solve()
+        self.node.submitblock(block.serialize().hex())
+
+        last_block_txs = self.node.getblock(self.node.getbestblockhash())['tx']
+        assert spend_even_tx.rehash() in last_block_txs
+        assert spend_odd_tx.rehash() in last_block_txs
+
+    def test_pubkeys_not_on_curve(self):
+        while True:
+            invalid_xonly_pub = random.randbytes(32)
+            if not GE.is_valid_x(int.from_bytes(invalid_xonly_pub, 'big')):
+                break
+
+        self.log.info("Check that creating P2PK outputs with not-on-curve pubkey is standard")
+        for compressed_prefix in [b'\x02', b'\x03']:
+            invalid_compressed_pub = compressed_prefix + invalid_xonly_pub
+            self.wallet.send_to(from_node=self.node, scriptPubKey=key_to_p2pk_script(invalid_compressed_pub), amount=21000)
+        invalid_uncompressed_pub = b'\x04' + invalid_xonly_pub + random.randbytes(32)
+        self.wallet.send_to(from_node=self.node, scriptPubKey=key_to_p2pk_script(invalid_uncompressed_pub), amount=21000)
+
+        self.log.info("Check that creating P2MS outputs with not-on-curve pubkeys is standard")
+        for n in (1, 2, 3):
+            p2ms_script = keys_to_multisig_script([invalid_uncompressed_pub]*n)
+            self.wallet.send_to(from_node=self.node, scriptPubKey=p2ms_script, amount=21000)
+
+        self.log.info("Check that creating P2TR outputs with not-on-curve pubkey is standard")
+        self.wallet.send_to(from_node=self.node, scriptPubKey=output_key_to_p2tr_script(invalid_xonly_pub), amount=21000)
+
+    def test_future_segwit_versions(self):
+        self.log.info("Check that creating future segwit outputs (2 <= version <= 16) is standard")
+        funding_txs = []
+        for future_ver in range(2, 16+1):
+            witness_program = random.randbytes(random.randrange(2, 40+1))
```

Comment:
Thinking out loud:
Initially thought of ensuring the edge cases (2 and 40) are explicitly added in addition to random program size selection, but this would probably be overkill.